src/components/document/annotation-layer.tsx

```diff
@@ -55,9 +31,20 @@ export const AnnotationLayer = observer(function AnnotationLayer({
   const [sourceTileId, setSourceTileId] = useState("");
   const [sourceObjectId, setSourceObjectId] = useState("");
   const [sourceObjectType, setSourceObjectType] = useState<string | undefined>();
+  const [mouseX, setMouseX] = useState<number | undefined>();
+  const [mouseY, setMouseY] = useState<number | undefined>();
+  const divRef = useRef<HTMLDivElement>();
   const ui = useUIStore();
   const tileApiInterface = useContext(TileApiInterfaceContext);
 
+  const handleMouseMove: MouseEventHandler<HTMLDivElement> = event => {
+    if (divRef.current) {
+      const bb = divRef.current.getBoundingClientRect();
+      setMouseX(event.clientX - bb.left);
+      setMouseY(event.clientY - bb.top);
+    }
+  };
+
```

Comment:
It probably isn't a problem but this might get slow with lots of annotations. Each mouse move is going to trigger a re-render. I'd guess React is smart enough that this won't be a problem.

If it is a problem, it could be improved by using a MobX object or storing the mouse position which this handler updates. Then that MobX object is passed to the PreviewArrow. This way only the PreviewArrow is re-rendered as the mouse is moved around.